pkgs/by-name/vt/vtsls/package.nix

```diff
@@ -0,0 +1,92 @@
+{
+  stdenv,
+  lib,
+  fetchFromGitHub,
+  nodejs_20,
+  git,
+  pnpm_8,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "vtsls";
+  version = "0.2.6";
+
+  src = fetchFromGitHub {
+    owner = "yioneko";
+    repo = "vtsls";
+    rev = "server-v${finalAttrs.version}";
+    hash = "sha256-HCi9WLh4IEfhgkQNUVk6IGkQfYagg805Rix78zG6xt0=";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [
+    nodejs_20
+    git
+    pnpm_8.configHook
+  ];
+
+  buildInputs = [ nodejs_20 ];
+
+  pnpmWorkspace = "@vtsls/language-server";
+
+  pnpmDeps = pnpm_8.fetchDeps {
+    inherit (finalAttrs)
+      pnpmWorkspace
+      pname
+      src
+      version
+      ;
+    hash = "sha256-4XxQ0Z2atTBItrD9iY7q5rJaCmb1EeDBvQ5+L3ceRXI=";
+  };
+
+  # Patches to get submodule sha from file instead of 'git submodule status'
+  patches = [ ./vtsls-build-patch.patch ];
+
+  # Skips manual confirmations during build
+  CI = true;
+
+  buildPhase = ''
+    runHook preBuild
+
+    # During build vtsls needs a working git installation.
+    git config --global user.name nixbld
+    git config --global user.email nixbld@example.com
+
+    # This is the sha for submodule packages/service/vscode. Unsure if there's a
+    # better way to get this.
+    echo "65d85f4cd0b4220fe8bb494898e001c7ad20e59b" > ./packages/service/HEAD
```

Comment:
Where is coming this ID from? This needs to be removed or extracted automatically.

Reply:
This is the SHA of the vscode submodule. I'll have to check if this is truly needed in our case. A SHA is used during `vtsls` build but I suspect it's just used to skip applying patches every time.

Reply:
Seems like using a dummy sha works just fine. Requires less patching of the build steps.